packages/protocol/contracts/layer1/based/TaikoInbox.sol

```diff
@@ -0,0 +1,704 @@
+        view
+        returns (Transition memory tran_)
+    {
+        Config memory config = getConfig();
+
+        uint64 slot = _batchId % config.batchRingBufferSize;
+        Batch storage batch = state.batches[slot];
+        require(batch.batchId == _batchId, BatchNotFound());
+
+        if (batch.verifiedTransitionId != 0) {
+            tran_ = state.transitions[slot][batch.verifiedTransitionId];
+        }
+    }
+
+    /// @inheritdoc ITaikoInbox
+    function getConfig() public view virtual returns (Config memory);
+
+    // Internal functions ----------------------------------------------------------------------
+
+    function __Taiko_init(
+        address _owner,
+        address _rollupResolver,
+        bytes32 _genesisBlockHash
+    )
+        internal
+    {
+        __Essential_init(_owner, _rollupResolver);
+
+        require(_genesisBlockHash != 0, InvalidGenesisBlockHash());
+        state.transitions[0][1].blockHash = _genesisBlockHash;
+
+        Batch storage batch = state.batches[0];
+        batch.metaHash = bytes32(uint256(1));
+        batch.lastBlockTimestamp = uint64(block.timestamp);
+        batch.anchorBlockId = uint64(block.number);
+        batch.nextTransitionId = 2;
+        batch.verifiedTransitionId = 1;
+
+        state.stats1.genesisHeight = uint64(block.number);
+
+        state.stats2.lastProposedIn = uint56(block.number);
+        state.stats2.numBatches = 1;
+
+        emit BatchesVerified(0, _genesisBlockHash);
+    }
+
+    function _unpause() internal override {
+        state.stats2.lastUnpausedAt = uint64(block.timestamp);
+        state.stats2.paused = false;
+    }
+
+    function _pause() internal override {
+        state.stats2.paused = true;
+    }
+
+    function _calcTxListHash(
+        uint8 _firstBlobIndex,
+        uint8 _numBlobs
+    )
+        internal
+        view
+        virtual
+        returns (bytes32)
+    {
+        bytes32[] memory blobHashes = new bytes32[](_numBlobs);
+        for (uint256 i; i < _numBlobs; ++i) {
+            blobHashes[i] = blobhash(_firstBlobIndex + i);
+            require(blobHashes[i] != 0, BlobNotFound());
+        }
+        return keccak256(abi.encode(blobHashes));
+    }
+
+    // Private functions -----------------------------------------------------------------------
+
+    function _verifyBatches(
+        Config memory _config,
+        Stats2 memory _stats2,
+        uint256 _length
+    )
+        private
+    {
+        uint64 batchId = _stats2.lastVerifiedBatchId;
+        uint256 slot = batchId % _config.batchRingBufferSize;
+        Batch storage batch = state.batches[slot];
+        uint24 tid = batch.verifiedTransitionId;
+        bytes32 blockHash = state.transitions[slot][tid].blockHash;
+
+        SyncBlock memory synced;
+
+        uint256 stopBatchId = (_config.maxBatchesToVerify * _length + _stats2.lastVerifiedBatchId)
+            .min(_stats2.numBatches);
+
+        for (++batchId; batchId < stopBatchId; ++batchId) {
```

Comment:
```suggestion
        for (++batchId; batchId <= stopBatchId; ++batchId) {
```

Reply:
If `lastVerifiedBatchId` is 0, the `batchId` is 1 and the `stopBatchId ` is 1 so the here should use `<=`.

Reply:
I see, `numBatches` already `+= 1` before `_verifyBatches` function, so `_config.maxBatchesToVerify * _length + _stats2.lastVerifiedBatchId` also change to `_config.maxBatchesToVerify * _length + _stats2.lastVerifiedBatchId + 1` to keep aligned with `numBatches`.